pkg/controller/collector/timestamp.go

```diff
@@ -57,43 +66,42 @@ func NewTimestamp(config TimestampConfig) (*Timestamp, error) {
 }
 
 func (t *Timestamp) Collect(ch chan<- prometheus.Metric) error {
-	ctx := context.Background()
-	gvk, err := apiutil.GVKForObject(t.newRuntimeObjectFunc(), t.k8sClient.Scheme())
-	if err != nil {
-		return microerror.Mask(err)
+	list := &metav1.PartialObjectMetadataList{}
+	{
+		gvk, err := apiutil.GVKForObject(t.newRuntimeObjectFunc(), t.scheme)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+		gvk.Kind = fmt.Sprintf("%sList", gvk.Kind)
+		list.SetGroupVersionKind(gvk)
```

Comment:
Are you sure you need to do all of that? I think `ctrlClient.List` will do that for you.

Reply:
I will investigate.

Reply:
Reading through the code, the client does strip off the `List` suffix from the `Kind` so there's no reason for line 75. I think we still need to to call `SetGroupVersionKind` because otherwise we'll get all objects that match the selector instead of just the kind we're interested in. I'll push a small fix taking this into account.

Reply:
Done.

Reply:
Actually, without that line I get an error saying `non-list type *v1.PartialObjectMetadataList (kind \"/v1, Kind=Pod\") passed as output` so I can't see any way to change the code here.

Reply:
Okay yeah this is dynamic stuff.